tests/test_internals.cc

```diff
@@ -0,0 +1,125 @@
+/**
+ * \file   test_internals.cc
+ * \author Lars Froehlich
+ * \date   Created on November 4, 2022
+ * \brief  Test suite for internal functions.
+ *
+ * \copyright Copyright 2022 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include <gul14/cat.h>
+#include <gul14/catch.h>
+
+#include "../src/internals.h"
```

Comment:
I usually do not like this kind of directory climbing, but the 'normal' solution (of adding an `-I` parameter to the compile command) is also not really nice; we do not want the internals be 'exposed' in all the other tests.